examples/javaweb-tomcat-sidecar/README.md

```diff
@@ -0,0 +1,216 @@
+[here](http://releases.k8s.io/release-1.0/examples/javaweb-tomcat-sidecar/README.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## Java Web Application with Tomcat and Sidercar Container
+
+The following document describes the deployment of a Java Web application using Tomcat. Instead of packaging `war` file inside the Tomcat image or mount the `war` as a volume, we use a sidecar container as `war` file provider.
+
+### Prerequisites
+
+https://github.com/kubernetes/kubernetes/blob/master/docs/user-guide/prereqs.md
+
+### Overview
+
+This sidecar mode brings a new workflow for Java users:
+
+![](workflow.png?raw=true "Workflow")
+
+As you can see, user can create a `sample:v2` container as sidecar to "provide" war file to Tomcat by copying it to the shared `emptyDir` volume. And Pod will make sure the two containers compose a "atomic" scheduling unit, which is perfect for this case. Thus, your application version management will be totally seperated from web server management.
+
+For example, if you gonna change the configurations of your Tomcat:
+
+```
+docker exec -it <tomcat_container_id> /bin/bash
+# do your change, and then commit it to a new image
+docker commit <tomcat_container_id> mytomcat:7.0-dev
+```
+
+Done! The new Tomcat image **will not** mess up with your `sample.war` file.
+
+You can also upgrade your app to new version seperately, without creating a new "Tomcat plus app" image (and this image will become huge as unionfs will keep all the old war files in its layers).
+
+**Why don't put my `sample.war` in a host dir and mount it to tomcat container?**
+
+You have to **manage the volumes** in this case, for example, when you restart or scale the Pod on another Node, your contents is not ready on that host.
+
+Generally, we have to set up a distributed file system (NFS at least) volume to solve this (if we do not have GCE PD volume). But seriously, it's a overkill.
+
+### How To
+
+In Kubernetes a [_Pod_](../../docs/user-guide/pods.md) is the smallest deployable unit that can be created, scheduled, and managed. Its a collocated group of containers that share an IP and storage volume.
+
+Here is the config [javaweb.yaml](javaweb.yaml) for Java Web pod:
+
+NOTE: you should define `war` contaienr **first** as it is the "provider".
+
+<!-- BEGIN MUNGE: javaweb.yaml -->
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: javaweb
+spec:
+  containers:
+  - image: resouer/sample:v1
+    name: war
+    volumeMounts:
+    - mountPath: /app
+      name: app-volume
+  - image: resouer/mytomcat:7.0
+    name: tomcat
+    command: ["sh","-c","/root/apache-tomcat-7.0.42-v2/bin/start.sh"]
+    volumeMounts:
+    - mountPath: /root/apache-tomcat-7.0.42-v2/webapps
+      name: app-volume
+    ports:
+    - containerPort: 8080
+      hostPort: 8001
+  volumes:
+  - name: app-volume
```

Comment:
I believe implicit emptyDir volumes are deprecated, and you should specify the source here (same below)